src/wallet/wallet.cpp

```diff
@@ -1425,11 +1419,11 @@ void CWallet::BlockConnected(const CBlock& block, const std::vector<CTransaction
     // the notification that the conflicted transaction was evicted.
 
     for (const CTransactionRef& ptx : vtxConflicted) {
-        SyncTransaction(ptx, {} /* block hash */, 0 /* position in block */);
+        SyncTransaction(ptx, CWalletTx::Status::CONFLICTED, {} /* block hash */, 0 /* position in block */);
```

Comment:
In commit "Encapsulate tx status in a Confirmation struct" (a31be09bfd77eed497a8e251d31358e16e2f2eb1)

This line is removed in the immediate next commit (https://github.com/bitcoin/bitcoin/pull/16624/commits/7e89994133725125dddbfa8d45484e3b9ed51c6e), but technically it has some bugs. Original intent of this code was only ever to send `-walletnotify`notifications, not to update transaction state, and the change should have used `UNCONFIRMED` instead of `CONFLICTED` to stay equivalent to previous code. Problems with the change:

- It was an undocumented change in behavior in what was supposed to be a refactoring commit
- It forgot to pass the block hash to `SyncTransaction` so wasn't properly setting a conflicted state. Instead it set an inconsistent state that would sometimes be treated as unconfirmed, sometimes conflicted.
- It would compound reorg problems wallet already has treating transactions that are no longer conflicted as conflicted, see [*bug-stale-conflicted*](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/Wallet-Transaction-Conflict-Tracking#bug-stale-conflicted).

In the future it would actually be possible to mark these transactions as conflicted instead of unconfirmed. Way this could be done is described [*idea-more-conflicts*](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/Wallet-Transaction-Conflict-Tracking#idea-more-conflicts).